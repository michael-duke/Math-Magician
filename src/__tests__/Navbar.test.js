const { render, screen } = require('@testing-library/react');
require('@testing-library/jest-dom');
const { BrowserRouter: Router } = require('react-router-dom');
const Navbar = require('../components/Navbar').default;

describe('Test for Navbar component', () => {
  test('Renders Navbar correctly', () => {
    const navbar = render(
      <Router>
        <Navbar />
      </Router>
    );
    expect(navbar).toMatchSnapshot();
  });

  test('check title exist', () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );
    expect(
      screen.getByRole('heading', { name: /Math Magician/i })
    ).toBeInTheDocument();
  });
});
