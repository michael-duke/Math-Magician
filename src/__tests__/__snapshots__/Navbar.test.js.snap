// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Navbar component Renders Navbar correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex flex-wrap gap-8 justify-between items-center mx-7 mt-3 sm:mx-0 sm:flex-col font-stange"
      >
        <div
          class="flex items-center gap-4"
        >
          <img
            alt="Math-logo"
            class="w-24 h-24 object-cover object-bottom rounded-lg"
            src="app-logo.jpg"
          />
          <h2
            class="text-6xl sm:text-5xl font-semibold text-blue-500/80 sl:w-min"
          >
            Math Magician
          </h2>
        </div>
        <ul
          class="flex sl:flex-col sm:gap-8"
        >
          <li
            class="px-8 text-2xl hover:text-magic-blue/90 border-white border-r-2 last:border-0"
          >
            <a
              aria-current="page"
              class="activeLink"
              exact="true"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="px-8 text-2xl hover:text-magic-blue/90 border-white border-r-2 last:border-0"
          >
            <a
              exact="true"
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="px-8 text-2xl hover:text-magic-blue/90 border-white border-r-2 last:border-0"
          >
            <a
              exact="true"
              href="/quote"
            >
              Quotes
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex flex-wrap gap-8 justify-between items-center mx-7 mt-3 sm:mx-0 sm:flex-col font-stange"
    >
      <div
        class="flex items-center gap-4"
      >
        <img
          alt="Math-logo"
          class="w-24 h-24 object-cover object-bottom rounded-lg"
          src="app-logo.jpg"
        />
        <h2
          class="text-6xl sm:text-5xl font-semibold text-blue-500/80 sl:w-min"
        >
          Math Magician
        </h2>
      </div>
      <ul
        class="flex sl:flex-col sm:gap-8"
      >
        <li
          class="px-8 text-2xl hover:text-magic-blue/90 border-white border-r-2 last:border-0"
        >
          <a
            aria-current="page"
            class="activeLink"
            exact="true"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="px-8 text-2xl hover:text-magic-blue/90 border-white border-r-2 last:border-0"
        >
          <a
            exact="true"
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li
          class="px-8 text-2xl hover:text-magic-blue/90 border-white border-r-2 last:border-0"
        >
          <a
            exact="true"
            href="/quote"
          >
            Quotes
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
